lib/src/router/normalized_path.dart

```diff
@@ -0,0 +1,92 @@
+import 'lru_cache.dart';
+
+/// Represents a URL path that has been normalized.
+///
+/// Normalization includes:
+/// - Resolving `.` and `..` segments.
+/// - Removing empty segments caused by multiple consecutive slashes.
+/// - Ensuring the path starts with a `/`.
+///
+/// Instances are interned using an LRU cache for efficiency, meaning identical
+/// normalized paths will often share the same object instance.
+class NormalizedPath {
+  static final _interned = LruCache<String, NormalizedPath>(10000);
```

Comment:
Not fully decided on how to expose configuration of the lru cache to the user.

Reply:
I need it in the factory ctor, so it has to be static (or be passed on every ctor call). I could make it settable, so you can replace it. I'm not fan of passing it into the ctor.

Reply:
I have made it public, and settable.